Cython/Utility/ObjectHandling.c

```diff
@@ -1499,38 +1488,29 @@ static CYTHON_INLINE PyObject *__Pyx__GetModuleGlobalName(PyObject *name)
 #endif
 {
     PyObject *result;
-#if CYTHON_COMPILING_IN_LIMITED_API
+#if CYTHON_COMPILING_IN_LIMITED_API || !CYTHON_USE_DICT_VERSIONS
     if (unlikely(!$module_cname)) {
         return NULL;
     }
```

Comment:
This looks to be broken (it isn't currently tested well because it's only tested in `fail_in_init_after_atexit` in `module_init.srctree` and srctree tests are currently run with Py_LIMITED_API set in the Limited API tests. So you only find out about it by enabling it more generally).

I think it probably just needs to raise `NameError` in this case.

Reply:
OK, let me look into it.

Reply:
I tried to running manually the test you mentioned, and I was not able to trigger the code branch we are dealing with. For the time being I'm setting NameError if the error indicator is not already set.